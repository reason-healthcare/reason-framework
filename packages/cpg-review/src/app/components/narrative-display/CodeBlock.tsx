import hljs from 'highlight.js'
import { FC, useEffect, useRef } from 'react'
import '@/styles/narrativeDisplay.css'

interface CodeBlock {
  code: string
}

const CodeBlock: FC<CodeBlock> = ({ code }) => {
  const codeRef = useRef(null)

  useEffect(() => {
    if (codeRef.current) {
      hljs.highlightElement(codeRef.current)
    }
  }, [code])

  return (
    <div className="code-container">
      <pre style={{ whiteSpace: 'break-spaces', wordBreak: 'break-word' }}>
        <code ref={codeRef}>{code}</code>
      </pre>
    </div>
  )
}

export default CodeBlock
