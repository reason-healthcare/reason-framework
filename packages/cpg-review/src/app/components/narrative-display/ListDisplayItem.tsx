interface ListDisplayProps {
  content: JSX.Element[] | undefined
  heading?: string | undefined
}

const ListDisplayItem = ({ content, heading }: ListDisplayProps) => {
  if (content) {
    return (
      <div className="list-items">
        {heading != null && (
          <>
            <span className="details-description">{heading}</span>
            <span>:</span>
          </>
        )}
        <ul>{content}</ul>
      </div>
    )
  }
  return <></>
}

export default ListDisplayItem
