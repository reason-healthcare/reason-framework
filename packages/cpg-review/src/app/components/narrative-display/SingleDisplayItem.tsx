interface SingleDisplayProps {
  content: string | JSX.Element | undefined
  heading?: string | undefined
}

const SingleDisplayItem = ({ content, heading }: SingleDisplayProps) => {
  if (content != null) {
    return (
      <div className="single-item">
        {heading != null && (
          <>
            <span className="narrative-description">{heading}</span>
            <span>: </span>
          </>
        )}
        <span>{content}</span>
      </div>
    )
  }
  return <></>
}

export default SingleDisplayItem
