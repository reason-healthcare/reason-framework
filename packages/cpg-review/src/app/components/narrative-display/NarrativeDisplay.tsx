import {
  formatProperty,
  formatResourceType,
  formatTitle,
  formatUrl,
  getNodeIdFromResource,
  is,
  notEmpty,
} from '../../helpers'
import { useLocation, useNavigate } from 'react-router-dom'
import { useEffect, useState } from 'react'
import { CloseOutlined } from '@ant-design/icons'
import type { RadioChangeEvent } from 'antd'
import { Radio } from 'antd'
import BrowserResolver from 'resolver/browser'
import CodeBlock from './CodeBlock'
import BackButton from '../BackButton'
import SingleDisplayItem from './SingleDisplayItem'
import { NodeContent } from '@/types/NodeProps'
import '@/styles/narrativeDisplay.css'

const META = [
  'id',
  'publisher',
  'title',
  'status',
  'date',
  'resourceType',
  'text',
  'meta',
  'url',
  'contact',
  'name',
  'version',
  'content',
  'mapping',
  'snapshot',
  'parameter',
  'jurisdiction',
  'count',
]

interface NarrativeDisplayProps {
  resolver: BrowserResolver | undefined
  setSelectedNode: React.Dispatch<React.SetStateAction<string | undefined>>
  narrativeContent?: NodeContent | undefined
}

const NarrativeDisplay = ({
  resolver,
  setSelectedNode,
  narrativeContent,
}: NarrativeDisplayProps) => {
  const [resource, setResource] = useState<
    fhir4.FhirResource | fhir4.PlanDefinitionAction | undefined
  >()
  const [cql, setCql] = useState<string | undefined>()
  const [partOfIdentifier, setPartOfIdentifier] = useState<string | undefined>()
  const [contentFormat, setContentFormat] = useState<'text' | 'json'>('text')

  const navigate = useNavigate()
  const path = useLocation().pathname
  useEffect(() => {
    setResource(undefined)
    setPartOfIdentifier(undefined)
    setCql(undefined)
    /** Narrative content is set from node select. Preferred to resolving a canonical or reference to the content considering Action nodes do not represent Canonical resources */
    if (narrativeContent != null) {
      const { resource, partOf } = narrativeContent
      if (partOf != null) {
        setPartOfIdentifier(partOf.url)
      }
      setResource(resource)
    } else if (resolver != null) {
      const reference = path.split('/').slice(-2).join('/')
      const rawResource = resolver.resolveReference(reference)
      if (is.ActivityDefinition(rawResource) || is.Questionnaire(rawResource)) {
        setSelectedNode(getNodeIdFromResource(rawResource))
      }
      if (is.Library(rawResource)) {
        const cql = rawResource.content?.find(
          (c) => c.contentType === 'text/cql'
        )?.data
        if (cql != null) {
          setCql(atob(cql))
        }
      } else {
        setCql(undefined)
      }
      if (rawResource != null) {
        setResource(rawResource)
      }
    }
  }, [path, narrativeContent])

  const handleNarrativeClose = () => {
    setSelectedNode(undefined)
  }

  const onContentFormatChange = (e: RadioChangeEvent) => {
    setContentFormat(e.target.value)
  }

  let resourceDisplay
  if (resource != null) {
    const formattedContent =
      contentFormat === 'text' ? (
        Object.entries(resource)
          .map((e: [string, any]) => {
            const [key, value] = e
            if (key === 'meta' && value.profile != null) {
              return formatProperty(
                value.profile,
                resolver,
                navigate,
                'meta.profile'
              )
            } else if (!META.includes(key)) {
              return formatProperty(value, resolver, navigate, key)
            }
          })
          .filter(notEmpty)
      ) : (
        <CodeBlock code={JSON.stringify(resource, null, 1)} />
      )

    resourceDisplay = (
      <div className="narrative-container-outer">
        <div className="narrative-container-inner">
          <h2>{`${formatResourceType(resource)}: ${formatTitle(resource)}`}</h2>
          {partOfIdentifier != null && (
            <SingleDisplayItem
              heading="Part Of"
              content={formatUrl(partOfIdentifier, resolver, navigate)}
            />
          )}
          {formattedContent}
          {cql != null && contentFormat === 'text' && <CodeBlock code={cql} />}
          <Radio.Group
            onChange={onContentFormatChange}
            value={contentFormat}
            optionType="button"
            buttonStyle="solid"
            style={{ marginTop: '3rem' }}
          >
            <Radio value={'text'}>Text</Radio>
            <Radio value={'json'}>JSON</Radio>
          </Radio.Group>
        </div>
      </div>
    )
  }

  return (
    <>
      <div className="buttons-container">
        <BackButton />
        <CloseOutlined onClick={handleNarrativeClose} />
      </div>
      {resourceDisplay != null ? (
        resourceDisplay
      ) : (
        <p>{`Unable to load ${path}`}</p>
      )}
    </>
  )
}

export default NarrativeDisplay
