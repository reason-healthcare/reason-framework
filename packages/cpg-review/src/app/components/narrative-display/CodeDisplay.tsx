import hljs from 'highlight.js'
import { FC, useEffect, useRef } from 'react'
import '@/styles/NarrativeDisplay.css'

interface CodeDisplay {
  cql: string
}

const CodeDisplay: FC<CodeDisplay> = ({ cql }) => {
  const codeRef = useRef(null)

  useEffect(() => {
    if (codeRef.current) {
      hljs.highlightElement(codeRef.current)
    }
  }, [cql])

  return (
    <div className="cql-container">
      <pre style={{ whiteSpace: 'break-spaces', wordBreak: 'break-word' }}>
        <code ref={codeRef}>{cql}</code>
      </pre>
    </div>
  )
}

export default CodeDisplay
