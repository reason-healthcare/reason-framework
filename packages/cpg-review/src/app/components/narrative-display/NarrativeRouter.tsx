import '@/styles/NarrativeDisplay.css'
import BrowserResolver from 'resolver/browser'
import { Route, Routes, useNavigate } from 'react-router-dom'
import { useEffect } from 'react'
import NarrativeDisplay from './NarrativeDisplay'
import { NodeData } from '../../types/NodeData'
import { is } from 'helpers'

interface NarrativeRouterProps {
  nodeData: NodeData | undefined
  resolver: BrowserResolver | undefined
  setSelectedNode: React.Dispatch<React.SetStateAction<string | undefined>>
}

const NarrativeRouter = ({
  nodeData,

  resolver,
  setSelectedNode,
}: NarrativeRouterProps) => {
  const navigate = useNavigate()

  useEffect(() => {
    if (is.FhirResource(nodeData?.nodeDetails)) {
      const {
        resourceType,
        id
      } = nodeData.nodeDetails
      navigate(`${resourceType}/${id}`)
    } else {
      navigate('/')
    }
  }, [nodeData])

  return (
    <div className="narrative-section">
      <Routes>
        <Route
          path="/"
          element={
            <NarrativeDisplay
              resolver={resolver}
              setSelectedNode={setSelectedNode}
              nodeDetails={nodeData}
            />
          }
        />
        <Route
          path="/:resourceType/:id"
          element={
            <NarrativeDisplay
              resolver={resolver}
              setSelectedNode={setSelectedNode}
            />
          }
        />
      </Routes>
    </div>
  )
}

export default NarrativeRouter
