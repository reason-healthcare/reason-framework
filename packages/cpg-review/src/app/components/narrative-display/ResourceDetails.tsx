import {
  formatProperty,
  formatResourceType,
  formatTitle,
  is,
  KnowledgeArtifact,
  notEmpty,
  TerminologyArtifact,
} from '../../helpers'
import { useLocation, useNavigate } from 'react-router-dom'
import BrowserResolver from 'resolver/browser'
import BackButton from '../BackButton'
import CodeDisplay from './CodeDisplay'
import '@/styles/NarrativeDisplay.css'
import { useEffect, useState } from 'react'

interface ResourceDetailsProps {
  resolver: BrowserResolver | undefined
  setSelected: React.Dispatch<React.SetStateAction<string | undefined>>
  nodeDetails?:
    | fhir4.PlanDefinition
    | fhir4.PlanDefinitionAction
    | fhir4.ActivityDefinition
    | undefined
}

const ResourceDetails = ({
  resolver,
  setSelected,
  nodeDetails,
}: ResourceDetailsProps) => {
  const [resource, setResource] = useState<
    | fhir4.StructureDefinition
    | KnowledgeArtifact
    | TerminologyArtifact
    | fhir4.PlanDefinitionAction
    | undefined
  >()
  const [cql, setCql] = useState<string | undefined>()
  const [displayJson, setDisplayJson] = useState<boolean>(false)

  const navigate = useNavigate()
  const path = useLocation().pathname
  useEffect(() => {
    if (nodeDetails != null) {
      if (is.KnowledgeArtifact(nodeDetails)) {
        delete nodeDetails.text
      }
      setResource(nodeDetails)
      setCql(undefined)
    } else if (resolver != null) {
      const reference = path.split('/').slice(-2).join('/')
      const rawResource = resolver.resolveReference(reference)
      if (
        is.PlanDefinition(rawResource) ||
        is.ActivityDefinition(rawResource)
      ) {
        setSelected(`definition-${rawResource.id}`)
      }
      if (
        is.KnowledgeArtifact(rawResource) ||
        is.StructureDefinition(rawResource) ||
        is.TerminologyArtifact(rawResource)
      ) {
        delete rawResource.text
        setResource(rawResource)
      }
      if (is.Library(rawResource)) {
        setCql(resolver.resolveCql(rawResource.id))
      } else {
        setCql(undefined)
      }
    }
  }, [path, nodeDetails])

  if (resource != null) {
    const meta = [
      'id',
      'publisher',
      'title',
      'status',
      'date',
      'resourceType',
      'text',
      'meta',
      'url',
      'contact',
      'name',
      'version',
      'content',
      'mapping',
      'snapshot',
      'parameter',
    ]

    const formatedProperties = Object.entries(resource)
      .map((e: [string, any]) => {
        const [k, v] = e
        if (!meta.includes(k)) {
          return formatProperty(v, resolver, navigate, k)
        }
      })
      .filter(notEmpty)

    return (
      <div>
        <h2>{`${formatResourceType(resource) ?? 'Action'}: ${formatTitle(
          resource
        )}`}</h2>
        {!displayJson && cql != null ? (
          <>
            {formatedProperties}
            <CodeDisplay cql={cql} />
          </>
        ) : displayJson ? (
          <CodeDisplay cql={JSON.stringify(resource, null, 1)} />
        ) : (
          <>{formatedProperties}</>
        )}
        <div
          className={
            path === '/' ? 'buttons-container center' : 'buttons-container'
          }
        >
          <BackButton />
          <button
            className="format-button"
            onClick={() => setDisplayJson(!displayJson)}
          >
            {displayJson ? 'Text' : 'JSON'}
          </button>
        </div>
      </div>
    )
  }

  return (
    <>
      <p>Unable to load details</p>
      <BackButton />
    </>
  )
}

export default ResourceDetails
