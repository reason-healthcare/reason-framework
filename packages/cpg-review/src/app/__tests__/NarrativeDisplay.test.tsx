import { fireEvent, render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import '../../jest-setup/matchMedia'
import '@testing-library/jest-dom'
import NarrativeDisplay from 'components/narrative-display/NarrativeDisplay'
import { NarrativeDisplayProps } from 'components/narrative-display/NarrativeDisplay'
import expectedNodes from './test-data/nodesExpected.json'
import resolverData from './test-data/resolverData.json'
import BrowserResolver from 'resolver/browser'
import React from 'react'
import {BrowserRouter} from 'react-router-dom'
import jest from 'jest-mock'

const node = expectedNodes.nodes.find(node => node.data.nodeData.nodeDetails.title === 'Order monitoring tests for antirheumatic drug therapy (Methotrexate).' && node.type === 'contentNode')
const props: NarrativeDisplayProps = {
  resolver: new BrowserResolver(JSON.stringify(resolverData)) as BrowserResolver,
  setSelectedNode: jest.fn(),
  nodeDetails: JSON.parse(JSON.stringify(node?.data.nodeData))
}

test('Renders display panel with PD action details as formatted text', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <NarrativeDisplay {...props}/>
      </BrowserRouter>
    )
    expect(asFragment()).toMatchSnapshot()
  })

test('Renders action details as JSON on toggle', async() => {
  const user = userEvent.setup()

  const { asFragment } = render(
    <BrowserRouter>
      <NarrativeDisplay {...props}/>
    </BrowserRouter>
  )

  await user.click(screen.getByText('JSON'))
  expect(asFragment()).toMatchSnapshot()
})