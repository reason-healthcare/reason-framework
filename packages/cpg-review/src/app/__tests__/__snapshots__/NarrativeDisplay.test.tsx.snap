// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders action details as JSON on toggle 1`] = `
<DocumentFragment>
  <div
    class="buttons-container"
  >
    <div />
    <span
      aria-label="close"
      class="anticon anticon-close"
      role="img"
      tabindex="-1"
    >
      <svg
        aria-hidden="true"
        data-icon="close"
        fill="currentColor"
        fill-rule="evenodd"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M799.86 166.31c.02 0 .04.02.08.06l57.69 57.7c.04.03.05.05.06.08a.12.12 0 010 .06c0 .03-.02.05-.06.09L569.93 512l287.7 287.7c.04.04.05.06.06.09a.12.12 0 010 .07c0 .02-.02.04-.06.08l-57.7 57.69c-.03.04-.05.05-.07.06a.12.12 0 01-.07 0c-.03 0-.05-.02-.09-.06L512 569.93l-287.7 287.7c-.04.04-.06.05-.09.06a.12.12 0 01-.07 0c-.02 0-.04-.02-.08-.06l-57.69-57.7c-.04-.03-.05-.05-.06-.07a.12.12 0 010-.07c0-.03.02-.05.06-.09L454.07 512l-287.7-287.7c-.04-.04-.05-.06-.06-.09a.12.12 0 010-.07c0-.02.02-.04.06-.08l57.7-57.69c.03-.04.05-.05.07-.06a.12.12 0 01.07 0c.03 0 .05.02.09.06L512 454.07l287.7-287.7c.04-.04.06-.05.09-.06a.12.12 0 01.07 0z"
        />
      </svg>
    </span>
  </div>
  <div
    class="narrative-container-outer"
  >
    <div
      class="narrative-container-inner"
    >
      <h2>
        Action: Order monitoring tests for antirheumatic drug therapy (Methotrexate).
      </h2>
      <div
        class="single-item"
      >
        <span
          class="narrative-description"
        >
          Part Of
        </span>
        <span>
          : 
        </span>
        <span>
          <a
            href="/PlanDefinition/MethotrexateMonitoringRecommendation"
          >
            Methotrexate Monitoring Recommendation
          </a>
        </span>
      </div>
      <div
        class="cql-container"
      >
        <pre
          style="white-space: break-spaces; word-break: break-word;"
        >
          <code
            class="hljs language-prolog"
            data-highlighted="yes"
          >
            {
 
            <span
              class="hljs-string"
            >
              "extension"
            </span>
            : [
  {
   
            <span
              class="hljs-string"
            >
              "url"
            </span>
            : 
            <span
              class="hljs-string"
            >
              "http://hl7.org/fhir/StructureDefinition/cqf-qualityOfEvidence"
            </span>
            ,
   
            <span
              class="hljs-string"
            >
              "valueCodeableConcept"
            </span>
            : {
    
            <span
              class="hljs-string"
            >
              "coding"
            </span>
            : [
     {
      
            <span
              class="hljs-string"
            >
              "system"
            </span>
            : 
            <span
              class="hljs-string"
            >
              "http://terminology.hl7.org/CodeSystem/evidence-quality"
            </span>
            ,
      
            <span
              class="hljs-string"
            >
              "code"
            </span>
            : 
            <span
              class="hljs-string"
            >
              "high"
            </span>
            
     }
    ]
   }
  },
  {
   
            <span
              class="hljs-string"
            >
              "url"
            </span>
            : 
            <span
              class="hljs-string"
            >
              "http://hl7.org/fhir/StructureDefinition/cqf-strengthOfRecommendation"
            </span>
            ,
   
            <span
              class="hljs-string"
            >
              "valueCodeableConcept"
            </span>
            : {
    
            <span
              class="hljs-string"
            >
              "coding"
            </span>
            : [
     {
      
            <span
              class="hljs-string"
            >
              "system"
            </span>
            : 
            <span
              class="hljs-string"
            >
              "http://terminology.hl7.org/CodeSystem/recommendation-strength"
            </span>
            ,
      
            <span
              class="hljs-string"
            >
              "code"
            </span>
            : 
            <span
              class="hljs-string"
            >
              "strong"
            </span>
            
     }
    ]
   }
  },
  {
   
            <span
              class="hljs-string"
            >
              "url"
            </span>
            : 
            <span
              class="hljs-string"
            >
              "http://hl7.org/fhir/uv/cpg/StructureDefinition/cpg-directionOfRecommendation"
            </span>
            ,
   
            <span
              class="hljs-string"
            >
              "valueCodeableConcept"
            </span>
            : {
    
            <span
              class="hljs-string"
            >
              "coding"
            </span>
            : [
     {
      
            <span
              class="hljs-string"
            >
              "system"
            </span>
            : 
            <span
              class="hljs-string"
            >
              "http://hl7.org/fhir/uv/cpg/CodeSystem/cpg-recommendation-direction"
            </span>
            ,
      
            <span
              class="hljs-string"
            >
              "code"
            </span>
            : 
            <span
              class="hljs-string"
            >
              "for"
            </span>
            
     }
    ]
   }
  }
 ],
 
            <span
              class="hljs-string"
            >
              "title"
            </span>
            : 
            <span
              class="hljs-string"
            >
              "Order monitoring tests for antirheumatic drug therapy (Methotrexate)."
            </span>
            ,
 
            <span
              class="hljs-string"
            >
              "description"
            </span>
            : 
            <span
              class="hljs-string"
            >
              "Order monitoring tests for antirheumatic drug therapy (Methotrexate)."
            </span>
            ,
 
            <span
              class="hljs-string"
            >
              "code"
            </span>
            : [
  {
   
            <span
              class="hljs-string"
            >
              "coding"
            </span>
            : [
    {
     
            <span
              class="hljs-string"
            >
              "system"
            </span>
            : 
            <span
              class="hljs-string"
            >
              "http://hl7.org/fhir/uv/cpg/CodeSystem/cpg-common-process"
            </span>
            ,
     
            <span
              class="hljs-string"
            >
              "code"
            </span>
            : 
            <span
              class="hljs-string"
            >
              "diagnostic-testing"
            </span>
            
    }
   ]
  }
 ],
 
            <span
              class="hljs-string"
            >
              "condition"
            </span>
            : [
  {
   
            <span
              class="hljs-string"
            >
              "kind"
            </span>
            : 
            <span
              class="hljs-string"
            >
              "applicability"
            </span>
            ,
   
            <span
              class="hljs-string"
            >
              "expression"
            </span>
            : {
    
            <span
              class="hljs-string"
            >
              "language"
            </span>
            : 
            <span
              class="hljs-string"
            >
              "text/cql-identifier"
            </span>
            ,
    
            <span
              class="hljs-string"
            >
              "expression"
            </span>
            : 
            <span
              class="hljs-string"
            >
              "Should order CBC if on Methotrexate therapy and missing test"
            </span>
            
   }
  }
 ],
 
            <span
              class="hljs-string"
            >
              "input"
            </span>
            : [
  {
   
            <span
              class="hljs-string"
            >
              "type"
            </span>
            : 
            <span
              class="hljs-string"
            >
              "Observation"
            </span>
            ,
   
            <span
              class="hljs-string"
            >
              "profile"
            </span>
            : [
    
            <span
              class="hljs-string"
            >
              "http://example.org/StructureDefinition/ActiveMethotrexateFeature"
            </span>
            
   ]
  },
  {
   
            <span
              class="hljs-string"
            >
              "type"
            </span>
            : 
            <span
              class="hljs-string"
            >
              "Observation"
            </span>
            ,
   
            <span
              class="hljs-string"
            >
              "profile"
            </span>
            : [
    
            <span
              class="hljs-string"
            >
              "http://example.org/StructureDefinition/LastCbcPanelReportDateFeature"
            </span>
            
   ]
  }
 ],
 
            <span
              class="hljs-string"
            >
              "definitionCanonical"
            </span>
            : 
            <span
              class="hljs-string"
            >
              "http://example.org/ActivityDefinition/OrderCBCActivity"
            </span>
            
}
          </code>
        </pre>
      </div>
      <div
        class="ant-radio-group ant-radio-group-solid css-dev-only-do-not-override-kghr11"
        style="margin-top: 3rem;"
      >
        <label
          class="ant-radio-button-wrapper css-dev-only-do-not-override-kghr11"
        >
          <span
            class="ant-radio-button"
          >
            <input
              checked=""
              class="ant-radio-button-input"
              type="radio"
              value="text"
            />
            <span
              class="ant-radio-button-inner"
            />
          </span>
          <span>
            Text
          </span>
        </label>
        <label
          class="ant-radio-button-wrapper ant-radio-button-wrapper-checked css-dev-only-do-not-override-kghr11"
        >
          <span
            class="ant-radio-button ant-radio-button-checked"
          >
            <input
              class="ant-radio-button-input"
              type="radio"
              value="json"
            />
            <span
              class="ant-radio-button-inner"
            />
          </span>
          <span>
            JSON
          </span>
        </label>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Renders display panel with PD action details as formatted text 1`] = `
<DocumentFragment>
  <div
    class="buttons-container"
  >
    <div />
    <span
      aria-label="close"
      class="anticon anticon-close"
      role="img"
      tabindex="-1"
    >
      <svg
        aria-hidden="true"
        data-icon="close"
        fill="currentColor"
        fill-rule="evenodd"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M799.86 166.31c.02 0 .04.02.08.06l57.69 57.7c.04.03.05.05.06.08a.12.12 0 010 .06c0 .03-.02.05-.06.09L569.93 512l287.7 287.7c.04.04.05.06.06.09a.12.12 0 010 .07c0 .02-.02.04-.06.08l-57.7 57.69c-.03.04-.05.05-.07.06a.12.12 0 01-.07 0c-.03 0-.05-.02-.09-.06L512 569.93l-287.7 287.7c-.04.04-.06.05-.09.06a.12.12 0 01-.07 0c-.02 0-.04-.02-.08-.06l-57.69-57.7c-.04-.03-.05-.05-.06-.07a.12.12 0 010-.07c0-.03.02-.05.06-.09L454.07 512l-287.7-287.7c-.04-.04-.05-.06-.06-.09a.12.12 0 010-.07c0-.02.02-.04.06-.08l57.7-57.69c.03-.04.05-.05.07-.06a.12.12 0 01.07 0c.03 0 .05.02.09.06L512 454.07l287.7-287.7c.04-.04.06-.05.09-.06a.12.12 0 01.07 0z"
        />
      </svg>
    </span>
  </div>
  <div
    class="narrative-container-outer"
  >
    <div
      class="narrative-container-inner"
    >
      <h2>
        Action: Order monitoring tests for antirheumatic drug therapy (Methotrexate).
      </h2>
      <div
        class="single-item"
      >
        <span
          class="narrative-description"
        >
          Part Of
        </span>
        <span>
          : 
        </span>
        <span>
          <a
            href="/PlanDefinition/MethotrexateMonitoringRecommendation"
          >
            Methotrexate Monitoring Recommendation
          </a>
        </span>
      </div>
      <div
        class="list-items"
      >
        <span
          class="narrative-description"
        >
          Extension
        </span>
        <span>
          :
        </span>
        <ul>
          <li>
            <div
              class="single-item"
            >
              <span>
                <span>
                  cqf-qualityOfEvidence
                </span>
                : 
                <span>
                  <span>
                    "high" from 
                  </span>
                  <span>
                    <a
                      href="http://terminology.hl7.org/CodeSystem/evidence-quality"
                      target="_blank"
                    >
                      http://terminology.hl7.org/CodeSystem/evidence-quality
                    </a>
                  </span>
                  <span />
                </span>
              </span>
            </div>
          </li>
          <li>
            <div
              class="single-item"
            >
              <span>
                <span>
                  cqf-strengthOfRecommendation
                </span>
                : 
                <span>
                  <span>
                    "strong" from 
                  </span>
                  <span>
                    <a
                      href="http://terminology.hl7.org/CodeSystem/recommendation-strength"
                      target="_blank"
                    >
                      http://terminology.hl7.org/CodeSystem/recommendation-strength
                    </a>
                  </span>
                  <span />
                </span>
              </span>
            </div>
          </li>
          <li>
            <div
              class="single-item"
            >
              <span>
                <span>
                  cpg-directionOfRecommendation
                </span>
                : 
                <span>
                  <span>
                    "for" from 
                  </span>
                  <span>
                    <a
                      href="http://hl7.org/fhir/uv/cpg/CodeSystem/cpg-recommendation-direction"
                      target="_blank"
                    >
                      http://hl7.org/fhir/uv/cpg/CodeSystem/cpg-recommendation-direction
                    </a>
                  </span>
                  <span />
                </span>
              </span>
            </div>
          </li>
        </ul>
      </div>
      <div
        class="single-item"
      >
        <span
          class="narrative-description"
        >
          Description
        </span>
        <span>
          : 
        </span>
        <span>
          Order monitoring tests for antirheumatic drug therapy (Methotrexate).
        </span>
      </div>
      <div
        class="single-item"
      >
        <span
          class="narrative-description"
        >
          Code
        </span>
        <span>
          : 
        </span>
        <span>
          <span>
            "diagnostic-testing" from 
          </span>
          <span>
            <a
              href="http://hl7.org/fhir/uv/cpg/CodeSystem/cpg-common-process"
              target="_blank"
            >
              http://hl7.org/fhir/uv/cpg/CodeSystem/cpg-common-process
            </a>
          </span>
          <span />
        </span>
      </div>
      <div
        class="single-item"
      >
        <span
          class="narrative-description"
        >
          Condition
        </span>
        <span>
          : 
        </span>
        <span>
          <span>
            <span>
              Should order CBC if on Methotrexate therapy and missing test as text/cql-identifier
            </span>
            <span />
          </span>
          <span>
             (applicability)
          </span>
        </span>
      </div>
      <div
        class="list-items"
      >
        <span
          class="narrative-description"
        >
          Input
        </span>
        <span>
          :
        </span>
        <ul>
          <li>
            <div
              class="single-item"
            >
              <span>
                Observation: 
                <span>
                  <a
                    href="/StructureDefinition/ActiveMethotrexateFeature"
                  >
                    Active Methotrexate Feature
                  </a>
                </span>
              </span>
            </div>
          </li>
          <li>
            <div
              class="single-item"
            >
              <span>
                Observation: 
                <span>
                  <a
                    href="/StructureDefinition/LastCbcPanelReportDateFeature"
                  >
                    Last CBC Panel Report Date Feature
                  </a>
                </span>
              </span>
            </div>
          </li>
        </ul>
      </div>
      <div
        class="single-item"
      >
        <span
          class="narrative-description"
        >
          Definition Canonical
        </span>
        <span>
          : 
        </span>
        <span>
          <a
            href="/ActivityDefinition/OrderCBCActivity"
          >
            Order CBC Activity
          </a>
        </span>
      </div>
      <div
        class="ant-radio-group ant-radio-group-solid css-dev-only-do-not-override-kghr11"
        style="margin-top: 3rem;"
      >
        <label
          class="ant-radio-button-wrapper ant-radio-button-wrapper-checked css-dev-only-do-not-override-kghr11"
        >
          <span
            class="ant-radio-button ant-radio-button-checked"
          >
            <input
              checked=""
              class="ant-radio-button-input"
              type="radio"
              value="text"
            />
            <span
              class="ant-radio-button-inner"
            />
          </span>
          <span>
            Text
          </span>
        </label>
        <label
          class="ant-radio-button-wrapper css-dev-only-do-not-override-kghr11"
        >
          <span
            class="ant-radio-button"
          >
            <input
              class="ant-radio-button-input"
              type="radio"
              value="json"
            />
            <span
              class="ant-radio-button-inner"
            />
          </span>
          <span>
            JSON
          </span>
        </label>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
